hail/python/hail/expr/types.py

```diff
@@ -789,6 +819,17 @@ def _convert_from_encoding(self, byte_reader, _should_freeze: bool = False) -> n
             np_type = self.element_type.to_numpy()
             return np.ndarray(shape=shape, buffer=np.array(elements, dtype=np_type), dtype=np_type, order="F")
 
+    def _convert_to_encoding(self, byte_writer, value: np.ndarray):
+        for dim in value.shape:
+            byte_writer.write_int64(dim)
+
+        if value.size > 0:
+            if self.element_type in _numeric_types:
+                byte_writer.write_bytes(value.data)
+            else:
+                for elem in np.nditer(value, order='F'):
+                    self.element_type._convert_to_encoding(byte_writer, elem)
```

Comment:
I feel good about this now. You're using `ENDArrayColumnMajor` and the column ordering is Fortran, so ti all checks out.